tunix/rl/grpo/grpo_learner.py

```diff
@@ -427,31 +427,37 @@ def train(
   ) -> None:
     """GRPO training loop.
 
-    Algorithm as below: extract from https://arxiv.org/abs/2402.03300
-    Input initial policy model πθinit; reward models rφ; task prompts D;
-    hyperparameters ε, β, μ
-
-    policy model πθ ← πθinit
-    for iteration = 1, ..., I do
-      reference model πref ← πθ
-      for step = 1, ..., M do
-        Sample a batch D♭ from D
-        Update the old policy model πθold ← πθ
-        Sample G outputs {oi}G_i=1 ~ πθold(· | q) for each question q ∈ D♭
-        Compute rewards {ri}G_i=1 for each sampled output oi by running rφ
-        Compute Âi,t for the t-th token of oi through group relative advantage
-        estimation.
-        for GRPO iteration = 1, ..., μ do
-          Update the policy model πθ by maximizing the GRPO objective (Equation
-          21)
-      Update rφ through continuous training using a replay mechanism.
-    Output πθ
-
-    NOTE:
-    1. The outer loop (I) is ignored for now because we never update the
-    reference model for now.
-    2. Currently sample and train hold the same referece to the model. So we
-    also omit the step to update the sampler model.
+    Algorithm as below: extract from https://arxiv.org/abs/2402.03300 ::
```

Comment:
ReST syntax: `::` here + indent the following paragraph to make it it lay out as a code-block.